src/dune_rules/context.ml

```diff
@@ -874,11 +874,10 @@ module DB = struct
     get context
 end
 
-let compiler t (mode : Lib_mode.t) =
+let compiler t (mode : Ocaml.Mode.t) =
   match mode with
-  | Ocaml Byte -> Ok t.ocamlc
-  | Ocaml Native -> t.ocamlopt
-  | Melange -> t.melc
```

Comment:
I don't think the `melc` field in the context record would be needed any longer.